src/network/networkd-dhcp-server.c

```diff
@@ -395,3 +404,169 @@ int config_parse_dhcp_server_emit(
                 emit->addresses[emit->n_addresses++] = a.in;
         }
 }
+
+int config_parse_dhcp_static_lease_address(
+        const char *unit,
+        const char *filename,
+        unsigned line,
+        const char *section,
+        unsigned section_line,
+        const char *lvalue,
+        int ltype,
+        const char *rvalue,
+        void *data,
+        void *userdata) {
+
+        assert(filename);
+        assert(lvalue);
+        assert(rvalue);
+        assert(data);
+
+        Network *network = data;
+        sd_dhcp_static_lease **current_lease_ref = &network->current_static_lease;
```

Comment:
We usually use `NetworkConfigSection` to manage which section we are now configuring. Please read e.g. networkd-address.c and .h.

Reply:
I am not sure I understand, `sd_dhcp_static_lease **current_lease_ref = &network->current_static_lease;` is not used to manage a section, but it is used to fill both `client_id` and `address` parts of the sd_dhcp_static_lease before putting them into `the static_leases_ref` hashmap. This is done because the section can be filled out correctly either like this:

``` 
[DHCPServerStaticLease]
MACAddress=12:34:56:78:9a:be
Address=10.1.1.4
```

or like this

```
[DHCPServerStaticLease]
Address=10.1.1.4
MACAddress=12:34:56:78:9a:be
```

So the static lease can be ready for placement in the hashmap either at the end of `config_parse_dhcp_static_lease_hwaddr` or at the end of `config_parse_dhcp_static_lease_address`.

Could You please explain a bit further how can I take advantage of the `NetworkConfigSection` for this use?

Reply:
The current code cannot handle e.g. the following setting:
```
[DHCPServerStaticLease]
MACAddress=12:34:56:78:9a:be
Address=10.1.1.4
MACAddress=12:34:56:78:9a:be
Address=10.1.1.5
MACAddress=12:34:56:78:9a:bc
```
This is the reason why we need to manage settings by its section, instead of `client_id`.

Reply:
I was not aware that the code is supposed to handle that setting. When the configuration was discussed multiple static leases were supposed to be in separate sections [e.g.](https://github.com/systemd/systemd/pull/15556#issuecomment-642529561) and the [response](https://github.com/systemd/systemd/pull/15556#issuecomment-644071637). Just to double-check is the code supposed to handle just the example setting You wrote, or both?

Reply:
The example I wrote above must be equal to the following.
```
[DHCPServerStaticLease]
Address=10.1.1.5
MACAddress=12:34:56:78:9a:bc
```
That is, only the last assignment should be used. The above example does not define multiple static leases.